source/material_model/depth_dependent.cc

```diff
@@ -0,0 +1,427 @@
+/*
+  Copyright (C) 2011 - 2015 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+#include <deal.II/base/std_cxx11/array.h>
+#include <aspect/material_model/depth_dependent.h>
+#include <deal.II/base/parameter_handler.h>
+#include <utility>
+#include <limits>
+
+using namespace dealii;
+
+namespace aspect
+{
+  namespace MaterialModel
+  {
+    template <int dim>
+    void
+    DepthDependent<dim>::read_viscosity_file(const std::string &filename)
+    {
+      /* This method is used for the Table method of depth dependent viscosity */
+      std::string temp;
+      std::ifstream in(filename.c_str(), std::ios::in);
+      AssertThrow (in,
+                   ExcMessage (std::string("Couldn't open file <") + filename + std::string(">")));
+
+      double min_depth=std::numeric_limits<double>::max();
+      double max_depth=std::numeric_limits<double>::lowest();
+
+      /* The input viscosity file has two columns that are the viscosity and Depth */
+      std::string header;
+      getline(in, header);/* Discard header line */
+      std::vector<double> visc_vec;
+      std::vector<double> depth_vec;
+      while (!in.eof())
+        {
+          double visc, depth;
+          in >> visc;
+          in >> depth;
+          if ( in.eof() ) break;
+          min_depth = std::min(depth, min_depth);
+          max_depth = std::max(depth, max_depth);
+
+          //        depth_vec.push_back( depth );
+          depth_table[0].push_back(depth);
+          visc_vec.push_back( visc );
+        }
+      /* Check to make sure that limits of depth-dependence table encompass entire domain. */
+      Assert( min_depth <= 0.0, ExcMessage("Minumum depth in viscosity file must be <= 0.0") );
+      Assert( max_depth >= this->get_geometry_model().maximal_depth(),
+              ExcMessage("Maximum depth in viscosity file must be >= maximal depth of model") );
+
+      viscosity_table.reset( new Table<1,double>( depth_table[0].size()) );
```

Comment:
you don't actually ever use the `viscosity_table` variable anywhere outside of this function. Instead of making it a pointer-valued member variable, just make it a table right here and now as a local variable.